diff --git a/fs/Kconfig b/fs/Kconfig
index 1c1ed36c..8fb40ecc 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -313,4 +313,11 @@ source "fs/nls/Kconfig"
 source "fs/dlm/Kconfig"
 source "fs/unicode/Kconfig"
 
+config NOMOUNT
+	bool "NoMount Path Redirection Subsystem"
+	default y
+	help
+	  NoMount allows path redirection and virtual file injection
+	  without mounting filesystems. Useful for systemless modifications.
+
 endmenu
diff --git a/fs/Makefile b/fs/Makefile
index 580e49be..4685ff21 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -20,6 +20,8 @@ else
 obj-y +=	no-block.o
 endif
 
+obj-$(CONFIG_NOMOUNT) += nomount.o
+
 obj-$(CONFIG_PROC_FS) += proc_namespace.o
 
 obj-y				+= notify/
